kerchunk/grib2.py

```diff
@@ -655,236 +649,6 @@ def parse_grib_idx(
-        The xarray datatree representation of the reference filesystem
-    kerchunk_store : dict
-        the grib_tree output for a single grib file
-    grib : bool
-        boolean for treating coordinates as grib levels
-
-    Returns
-    -------
-    pandas.Dataframe : The dataframe constructed from the grib metadata
-    """
-    import pandas as pd
-
-    result: list[dict] = []
-
-    for node in dtree.subtree:
-        if node.has_data:
-            result += extract_dataset_chunk_index(
-                node, kerchunk_store["refs"], grib=grib
-            )
-
-    return pd.DataFrame.from_records(result)
-
-
-def _map_grib_file_by_group(
-    fname: str,
-    mapper: Optional[Callable] = None,
-    storage_options: Optional[Dict] = None,
-) -> "pd.DataFrame":
-    """
-    Helper method used to read the cfgrib metadata associated with each message (group) in the grib file
-    This method does not add metadata
-
-    Parameters
-    ----------
-    fname : str
-        the file name to read with scan_grib
-    mapper : Optional[Callable]
-        the mapper if any to apply (used for hrrr subhf)
-
-    Returns
-    -------
-    pandas.Dataframe : The intermediate dataframe constructed from the grib metadata
-    """
-    import pandas as pd
-
-    mapper = (lambda x: x) if mapper is None else mapper
-    references = scan_grib(fname, storage_options=storage_options)
-
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore")
-        return pd.concat(
-            # grib idx is fortran indexed (from one not zero)
-            list(
-                filter(
-                    # filtering out the cfgrib metadata dataframe in case it is None
-                    lambda item: item is not None,
-                    [
-                        # extracting the metadata from a single message
-                        _extract_single_group(mapper(group), i, storage_options)
-                        for i, group in enumerate(references, start=1)
-                    ],
-                )
-            )
-        ).set_index("idx")
-
-
-def _extract_single_group(grib_group: dict, idx: int, storage_options: Dict):
-    import datatree
-
-    grib_tree_store = grib_tree(
-        [
-            grib_group,
-        ],
-        storage_options,
-    )
-
-    if len(grib_tree_store["refs"]) <= 1:
-        logger.info("Empty DT: %s", grib_tree_store)
-        return None
-
-    dt = datatree.open_datatree(
-        fsspec.filesystem("reference", fo=grib_tree_store).get_mapper(""),
-        engine="zarr",
-        consolidated=False,
-    )
-
-    k_ind = extract_datatree_chunk_index(dt, grib_tree_store, grib=True)
-    if k_ind.empty:
-        logger.warning("Empty Kind: %s", grib_tree_store)
-        return None
-
-    assert (
-        len(k_ind) == 1
-    ), f"expected a single variable grib group but produced: {k_ind}"
-    k_ind.loc[:, "idx"] = idx
-    return k_ind
-
-
 def build_idx_grib_mapping(
```

Comment:
Why did this function not move? I would think it should live in `_grib_index`, but it can be _imported_ here, to indicate that, yes, it is user-facing.